feat: Cross-track upgrade documentation (#174)
canonical/observability-stack: docs/how-to/deploy-and-manage/upgrade.md

Author: We want to ensure that the user gets the latest TF module with `-upgrade`.
```suggestion
    terraform init -upgrade; terraform apply
```

Author: ```suggestion
2. Manually refresh all charms to the latest revision in 2/stable 
```

Author: ```suggestion
1. Update the channel input to track 2/stable and then:
```

Author: ```suggestion
4. Once Terraform has applied all resources, apply again until no new resources are applied:
    ```bash
    terraform apply
    ```

### Migrate from COS Lite 2 to COS Lite 3
```

### Migrate from COS 2 to COS 3
Using Terraform:

1. Update the channel input to track 2/stable and then:
    ```bash
    terraform init -upgrade; terraform apply
    ```

> 2. Manually refresh all charms to the latest revision in 2/stable 

Author: This is not needed, but helps to give more upgrade guarantee for deployments which use old revisions in track

    ```bash
    juju refresh <charm-name> --channel 2/stable
    ```
3. Update the Terraform module source ref to tf-cos-3.0.0 and then:
    ```bash
    terraform init -upgrade; terraform apply
    ```
4. Once Terraform has applied all resources, apply again until no new resources are applied:
    ```bash
    terraform apply
    ```

### Migrate from COS Lite 2 to COS Lite 3
Using Terraform:

1. Update the channel input to track 2/stable and then:
    ```bash
    terraform init -upgrade; terraform apply
    ```

> 2. Manually refresh all charms to the latest revision in 2/stable 

Author: This is not needed, but helps to give more upgrade guarantee for deployments which use old revisions in track

    ```bash
    juju refresh <charm-name> --channel 2/stable
    ```

> 3. Update all references to track `tf-cos-3.0.0` and then:

Author: ```suggestion
3. Update the Terraform module source ref to tf-cos-3.0.0 and then:
```

    ```bash

>     terraform apply

Author: ```suggestion
    terraform init -upgrade; terraform apply
```
